src/layer/LayersFactory.ts

```diff
@@ -194,18 +196,26 @@ export class LayersFactory {
     filterLayers: Function | null,
   ): Promise<AbstractLayer[]> {
     const parsedXml = await fetchGetCapabilitiesXml(baseUrl);
+
     const layersInfos = parsedXml.WMS_Capabilities.Capability[0].Layer[0].Layer.map(layerInfo => ({
       layerId: layerInfo.Name[0],
       title: layerInfo.Title[0],
       description: layerInfo.Abstract ? layerInfo.Abstract[0] : null,
       dataset: null,
+      legendUrl:
```

Comment:
Could this also be done in the `WmsLayer.ts`, similarly to `AbstractSentinelHubV1OrV2Layer.ts`? (in case you want to create a WMS layer manually and you want to have the legend)

Reply:
I'm sorry, but I don't quite understand what you mean.

Reply:
in `AbstractSentinelHubV1OrV2Layer.ts` you added the function `updateLayerFromServiceIfNeeded` with the code that fetches the legend url if you (manually) create a layer that extends `AbstractSentinelHubV1OrV2Layer` without the legend url.

Currently you can manually create a `WmsLayer` without legend url but there is no way to update the `WmsLayer` with additional properties, for example the legend url (no `updateLayerFromServiceIfNeeded`).

The code of `updateLayerFromServiceIfNeeded` in `WmsLayer` would be the same as in `AbstractSentinelHubV1OrV2Layer` because the OGC `GetCapabilities` is in XML format.

Reply:
You are correct. 

However, - correct me if I'm wrong - I don't think you can update any other info for any other layer either, except for those specific parameters in `updateLayerFromServiceIfNeeded`. For example, if one creates a layer, I don't think they can get the `description` from service? I might be wrong. This should probably be implemented?

The slight issue I have with this is, that `updateLayerFromServiceIfNeeded` generally fetches data from a different endpoint that `makeLayers` use. However in this case, it would be exactly the same. So `makeLayer(instanceId, layerId).updateLayerFromServiceIfNeeded()` will basically do the same thing twice for a `WMSLayer`. 

Additionally, I don't think we should override the manually set values.

Therefore, I decided *not* to update if `legendUrl` is already defined.

Reply:
> However, - correct me if I'm wrong - I don't think you can update any other info for any other layer either, except for those specific parameters in updateLayerFromServiceIfNeeded. 

You are correct. 

> For example, if one creates a layer, I don't think they can get the description from service? I might be wrong. This should probably be implemented?

I think they can get the description from the service in any other way than requesting `getCapabilities`. For Sentinel Hub layers, it's only possible if you have the instance id ([docs](https://docs.sentinel-hub.com/api/latest/reference/#operation/getWmsLayers)). I don't think we need getting the layer description at this moment.

> Therefore, I decided not to update if legendUrl is already defined.

:+1: I think it's great like that.